### PR TITLE
FEATURE: Provide Docker Compose setup for development

### PR DESCRIPTION
**Pull request by robertlemke**
This is a first version of a Docker Compose setup
in order to develop neos.io. Some of the packages currently
used in the docker-compose.yml are not yet accessible for
everyone, but will be in the near future. They are those
images used in production for www.neos.io (apart from
Elasticsearch).

### CODE REVIEW

File: docker-compose-up.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+if [ ! -d "Packages" ]; then
+  composer install
+fi
+
+docker-compose up -d --remove-orphans
+docker cp . "${BEACH_DEV_PROJECT_NAME}_php_1:/application/"
+docker exec "${BEACH_DEV_PROJECT_NAME}_php_1" /bin/bash -c "chown -R beach:beach /application"
+docker exec "${BEACH_DEV_PROJECT_NAME}_sidecar_1" /bin/bash -c "cd /application; ./flow doctrine:migrate"
+
+open ${BEACH_SITE_BASE_URI}
```
Comment (kdambekalns) [inline docker-compose-up.sh]: Does `open` usually exist on other OSs?

Comment (dfeyer) [inline docker-compose-up.sh]: It's pure OSX ;)

**Comment by kdambekalns** [comment]
So, make it `open ${BEACH_SITE_URI} || echo "Open ${BEACH_SITE_URI} in a browser"` instead?

File: docker-compose-up.sh (hunk 1 above)
Comment (kdambekalns) [inline docker-compose-up.sh]: So, make it `open ${BEACH_SITE_URI} || echo "Open ${BEACH_SITE_URI} in a browser"` instead?

Comment (robertlemke) [inline docker-compose-up.sh]: I'm totally open to that.

**Pull request merged**